swayfmt/src/utils/map/comments.rs

```diff
@@ -50,10 +50,14 @@ pub fn comment_map_from_src(input: Arc<str>) -> Result<(CommentMap, usize), Form
 
     // pass the input through lexer
     let mut start = 0;
-    if input.starts_with('\n') {
-        let mut input_chars = input.chars().peekable();
-        while let Some('\n') = input_chars.next() {
-            start += 1
+    for (char_index, curr_char) in input.chars().enumerate() {
```

Comment:
If you are trimming the `src` I'm not sure there's a need for this anymore

Reply:
Also can remove the `.trim_start()` from `newline.rs` and the logic from my previous PR #2844

Reply:
My attempt was to try and keep it as in tact as possible but your approach seems better

Reply:
Thanks for pointing out!